Source/Evolve/WarpXEvolve.cpp

```diff
@@ -351,12 +337,13 @@ WarpX::Evolve (int numsteps)
         }
 
         bool const do_diagnostic = (multi_diags->DoComputeAndPack(step) || reduced_diags->DoDiags(step));
+        bool const end_of_step_loop = (step == numsteps_max - 1) || (cur_time + dt[0] >= stop_time - 1.e-3*dt[0]);
```

Comment:
Just a question to clarify, where does the `1.e-3` factor come from here?

Reply:
It's a somewhat arbitrary factor to allow for round off error in the time. If the `cur_time + dt` is almost but not quite equal to `stop_time`, then that's close enough.